include/storage/io_config.hpp

```diff
@@ -0,0 +1,82 @@
+#ifndef IO_CONFIG_HPP
+#define IO_CONFIG_HPP
+
+#include <boost/filesystem/path.hpp>
+
+namespace osrm
+{
+namespace storage
+{
+struct IOConfig
+{
+    IOConfig() = default;
+
+    IOConfig(const boost::filesystem::path &base) : osrm_path(base) { UseDefaultOutputNames(); }
+
+    // Infer the output names from the path of the .osrm file
+    void UseDefaultOutputNames()
+    {
+        ram_index_path = {osrm_path.string() + ".ramIndex"};
+        file_index_path = {osrm_path.string() + ".fileIndex"};
+        hsgr_data_path = {osrm_path.string() + ".hsgr"};
+        nodes_data_path = {osrm_path.string() + ".nodes"};
+        edges_data_path = {osrm_path.string() + ".edges"};
+        core_data_path = {osrm_path.string() + ".core"};
+        geometries_path = {osrm_path.string() + ".geometry"};
+        timestamp_path = {osrm_path.string() + ".timestamp"};
+        turn_weight_penalties_path = {osrm_path.string() + ".turn_weight_penalties"};
+        turn_duration_penalties_path = {osrm_path.string() + ".turn_duration_penalties"};
+        turn_penalties_index_path = {osrm_path.string() + ".turn_penalties_index"};
+        datasource_names_path = {osrm_path.string() + ".datasource_names"};
+        names_data_path = {osrm_path.string() + ".names"};
+        properties_path = {osrm_path.string() + ".properties"};
+        intersection_class_path = {osrm_path.string() + ".icd"};
+        turn_lane_data_path = {osrm_path.string() + ".tld"};
+        turn_lane_description_path = {osrm_path.string() + ".tls"};
+        mld_partition_path = {osrm_path.string() + ".partition"};
+        mld_storage_path = {osrm_path.string() + ".cells"};
+        mld_graph_path = {osrm_path.string() + ".mldgr"};
+        level_path = {osrm_path.string() + ".level"};
+        node_path = {osrm_path.string() + ".enw"};
+        edge_based_graph_path = {osrm_path.string() + ".ebg"};
+        compressed_node_based_graph_path = {osrm_path.string() + ".cnbg"};
+        cnbg_ebg_mapping_path = {osrm_path.string() + ".cnbg_to_ebg"};
+        restriction_path = {osrm_path.string() + ".restrictions"};
+        intersection_class_data_path = {osrm_path.string() + ".icd"};
```

Comment:
Hm is there a way we can make it so that we get a warning if we forget to initialize a path below?

Should this e.g. all be done in the constructor, as in
```c++
struct io {
  io(path base)
    : a(base + '.a'),
      b(base + '.b') { }

  path a;
  path b;
};
```
?